src/v/cluster/archival/archival_metadata_stm.cc

```diff
@@ -488,7 +489,7 @@ ss::future<std::error_code> command_batch_builder::replicate() {
     auto batch = std::move(_builder).build();
     auto f = _stm.get()
                .do_replicate_commands(std::move(batch), _as)
-               .finally([u = std::move(units), h = _stm.get()._gate.hold()] {});
+               .finally([u = std::move(units), h = std::move(holder)] {});
```

Comment:
Why isn't it sufficient to just hold the gates for this entire `replicate()` call?

Reply:
Ah indeed, thanks!